File: c++/0387.cpp
// Link: https://leetcode.com/problems/first-unique-character-in-a-string/description/
// Runtime: 17 ms
// Memory: 10.8 MB

class Solution 
{
public:
    int firstUniqChar(const std::string& s) 
    {
        std::vector<int> symbolAndFrequency(26);
        for (const auto& symbol : s)
            symbolAndFrequency[symbol - 'a']++;
        
        for (int i = 0; i < static_cast<int>(s.size()); i++)
            if (symbolAndFrequency[s[i] - 'a'] == 1)
                return i;

        return -1;
    }
};

File: python/0387.py
# Link: https://leetcode.com/problems/first-unique-character-in-a-string/description/
# Runtime: 77 ms
# Memory: 16.75 MB


class Solution:
    def firstUniqChar(self, s: str) -> int:
        symbol_and_frequency = Counter(s)
        for index, symbol in enumerate(s):
            if symbol_and_frequency[symbol] == 1:
                return index

        return -1
